homeassistant/helpers/template.py

```diff
@@ -319,6 +319,12 @@ def __str__(self) -> str:
 RESULT_WRAPPERS[tuple] = TupleWrapper
 
 
+class RenderedTemplateResultStrWrapper(str):
+    """Wrapper class to store template result string to avoid double renders."""
+
+    __slots__ = ()
+
+
```

Comment:
Creating a subclass of `str` to allow the type-checking allows the existing code base to gracefully handle the change of return type (unless there are explicit equality checks of `kls == str` anywhere in the codebase instead of using `isinstance`).